braze/client.py

```diff
@@ -435,3 +488,60 @@ def unsubscribe_user_email(
         }
 
         return self._post_request(payload, BrazeAPIEndpoints.UNSUBSCRIBE_USER_EMAIL)
+
+    def retrieve_unsubscribed_emails(
+            self,
+            start_date,
+            end_date,
+            limit=100,
+            offset=0,
+            sort_direction='desc',
+    ):
+        """
+        Retrieve unsubscribe users email via API.
+
+        https://www.braze.com/docs/api/endpoints/email/get_query_unsubscribed_email_addresses/
+
+        Arguments:
+            start_date(str): Start date of the range to retrieve unsubscribes, must be earlier than end_date.
+            This is treated as midnight in UTC time by the API. Format: YYYY-MM-DD
+            end_date(str): End date of the range to retrieve unsubscribes. This is treated as midnight in
+            UTC time by the API. Format: YYYY-MM-DD
+            limit(int): To limit the number of results returned. Defaults to 100, maximum is 500.
+            offset(int): The beginning point in the list to retrieve from.
+            sort_direction(str): Pass in the value asc to sort unsubscribes from oldest to newest. Pass in desc to
+            sort from newest to oldest. If sort_direction is not included, the default order is newest to oldest.
+        Returns:
+            response (list): list of emails
+        """
+        try:
+            datetime.datetime.strptime(start_date, "%Y-%m-%d")
+            datetime.datetime.strptime(end_date, "%Y-%m-%d")
+        except ValueError as e:
+            raise BrazeClientError("Invalid date format. Please provide dates in YYYY-MM-DD format.") from e
+
+        params = {
+            'start_date': start_date,
+            'end_date': end_date,
+            'limit': limit,
+            'offset': offset,
+            'sort_direction': sort_direction,
+        }
+
+        unsubscribed_emails = []
+        # NOTE: If your date range has more than limit number of unsubscribes, you will need to make multiple API calls,
+        # each time increasing the offset until a call returns either fewer than limit or zero results.
+        while True:
+            response = self._get_request(params, BrazeAPIEndpoints.UNSUBSCRIBED_EMAILS)
+            emails = response.get('emails', [])
+            unsubscribed_emails.extend(emails)
+
+            # Check if limit is reached or no more emails available
+            if limit is None or len(emails) < limit or len(emails) == 0:
```

Comment:
`len(emails) == 0` is not needed since it `0 < limit` holds true here.

Reply:
As mentioned in the API documentation, if no results are obtained from the API, I have implemented a check to handle this scenario.

Reply:
yes `if no results are obtained` will also be handled with `len(emails) < limit` check. there is no need for an additional check.

Reply:
To explicitly check this scenario.

Reply:
Yep, you are right. but if this scenario is being handled by a previous condition, there is no need for an additional condition. let's keep the useful code only, that condition will never be reached.